### ISSUE TITLE
Route Viewer need to be closed 2 times if a route is not loaded

### ISSUE THREAD

**Issue by Kenny-Hui** [leezer3/OpenBVE#583]
### Description

Have to trace all the way back to 1.5.0.10 (At least that's the oldest version of this port I have)
Without loading any route, clicking the X Button once will not respond
Only the second time will actually close the window, this does not happen when a route is loaded or on 1.4.3.1
**Note:** On 1.7.2.0 or above, closing route viewer when a route is loading will cancel the route loading, intentional?

Expected behavior: Route Viewer should be closed by clicking the X button 1 time when a route is not loaded

### Reproduction
Open Route Viewer, don't load a route and try closing it. You will need to close 2 times

### Related information 
* Operating system: Windows 10 20H2
* Method of control: Keyboard

**Comment by leezer3** [comment]
The cancellation was deliberate, but the double-close wasn't.
I've fixed the double-close.

TLDR:
Loading a routefile is done in a separate thread (and now by an entirely separate plugin for that matter).
This means that we need to gracefully cleanup / terminate this thread in order to stop random Windows error messages if we terminate the thread unexpectedly.
